### PR TITLE
Support external mev builder

### PR DESCRIPTION
**Pull request by domiwei**
- enable builder mode by configuring `--beacon.api=builder` and setup `--mev-relay-url` to run in builder mode ([code](https://github.com/ledgerwatch/erigon/pull/10492/files#diff-aca4b9462813c7b1d2ff5d7929408096522e963a92a092fcb1ffc3240c9e43b9R111)) 

- Set mev relay url by `--caplin.mev-relay-url` as run erigon if would like to run caplin in builder mode. 
eg:
```
--caplin.mev-relay-url=https://boost-relay-sepolia.flashbots.net/
```
-  Follow builder specs [here](https://ethereum.github.io/builder-specs/#/) to impl [builder client](https://github.com/ledgerwatch/erigon/pull/10492/files#diff-d2932cc9922a090127fd132ae180802b3686ab83e26f1946c985f4c0da6ffc23R12)

- validator registration api `/v1/validator/register_validator`

- choose proposed block (either blinded or unblinded block) api `/v3/validator/blocks/{slot}`

- post blinded block api `/v1/beacon/blinded_blocks` `/v2/beacon/blinded_blocks`

### CODE REVIEW

**Comment by Giulio2002** [comment]
Ok will test

**Comment by Giulio2002** [comment]
```flag provided but not defined: -mev-relay-url```

did you add this to Erigon too?

File: cmd/caplin/caplinflags/flags.go [hunk 1]
```diff
@@ -108,4 +109,9 @@ var (
 		Usage: "Path to the token that ensures safe connection between CL and EL",
 		Value: "",
 	}
+	MevRelayUrl = cli.StringFlag{
```
[the remark this answers is not in the document]
Comment (domiwei) [reply cmd/caplin/caplinflags/flags.go:112]: done. will see log `Builder client is ready` if the flag is set.
`--caplin.mev-relay-url=https://boost-relay-sepolia.flashbots.net/`

File: cl/beacon/builder/client.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
+package builder
```
Comment (Giulio2002) [inline cl/beacon/builder/client.go:1]: All good, some unit tests would be awesome, you can probably test by making an http test server which mocks the actual real-life behaviour

File: cl/beacon/builder/types.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+	"github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon-lib/log/v3"
+	"github.com/ledgerwatch/erigon/cl/cltypes"
+	"github.com/ledgerwatch/erigon/cl/cltypes/solid"
+)
+
+type ExecutionPayloadHeader struct {
+	Version string `json:"version"`
+	Data    struct {
```
Comment (Giulio2002) [inline cl/beacon/builder/types.go:15]: Let's define many structs rather than declearing all of them at once

File: cl/cltypes/builder.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package cltypes
+
+import libcommon "github.com/ledgerwatch/erigon-lib/common"
+
+type ValidatorRegistration struct {
+	Message struct {
```
Comment (Giulio2002) [inline cl/cltypes/builder.go:6]: please separrate this struct into its own type

File: turbo/engineapi/engine_types/jsonrpc.go [hunk 5]
```diff
@@ -68,6 +69,24 @@ type BlobsBundleV1 struct {
 	Blobs       []hexutility.Bytes `json:"blobs"       gencodec:"required"`
 }
 
+func (b *BlobsBundleV1) Check() error {
```
Comment (Giulio2002) [inline turbo/engineapi/engine_types/jsonrpc.go:72]: Why is this needed?

**Review by Giulio2002 — CHANGES_REQUESTED** [review]

File: turbo/engineapi/engine_types/jsonrpc.go (hunk 5 above)
Comment (domiwei) [reply turbo/engineapi/engine_types/jsonrpc.go:72]: just simply check if content length is reasonable or not. maybe move it out from structure definition.

File: cl/beacon/builder/client.go (hunk 2 above)
Comment (domiwei) [reply cl/beacon/builder/client.go:1]: done by overwrite roundtripper

**Review by Giulio2002 — APPROVED** [review]
